helper/firmware_catalog_helper.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://mozilla.org/MPL/2.0/
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	"terraform-provider-ome/clients"
+	"terraform-provider-ome/models"
+	"terraform-provider-ome/utils"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+)
+
+// GetAllCatalogFirmware get all catalog firmware
+func GetAllCatalogFirmware(client *clients.Client) (*models.Catalogs, error) {
```

Comment:
This function seems redundant. Please check and remove if so.

Reply:
This is a technique we used in other providers. We write the code like this, so it is easy to mock the data in the tests in order to return error cases in the datasource/resource itself. Otherwise it is much more difficult to get simulate these error cases.

Reply:
Okay, makes sense. Since the clients can be potentially an external library, better to mock an internal wrapper. Thanks for the explanation!